application/src/main/java/org/togetherjava/tjbot/util/PresentationUtils.java

```diff
@@ -0,0 +1,40 @@
+package org.togetherjava.tjbot.util;
+
+import com.github.freva.asciitable.AsciiTable;
+import com.github.freva.asciitable.Column;
+import com.github.freva.asciitable.ColumnData;
+import com.github.freva.asciitable.HorizontalAlign;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+/**
+ * Utility class for representation of data in various formats
+ */
+public final class PresentationUtils {
+    private PresentationUtils() {}
+
+    /**
+     * Flattens a dataFrame to String representation of a table.
+     * 
+     * @param dataFrame dataframe represented as List<List<String>> where List<String> represents a
+     *        single row
+     * @param headers column headers for the table
+     * @param horizontalAligns column alignment for the table
+     * @return String representation of the dataFrame in tabular form
+     */
+    public static String dataFrameToAsciiTable(List<List<String>> dataFrame, String[] headers,
+            HorizontalAlign[] horizontalAligns) {
+        Objects.requireNonNull(dataFrame, "DataFrame cannot be null");
```

Comment:
Use the @NotNull annotation from Jetbrains
And add to the docs that it has to be non-null

Reply:
This I'll check but I mostly tend to rely on JDK libs if they already provide something similar.

Reply:
Annotations exist for a reason, and Objects#requireNonNull lacks of a lot.

Documentation state nothing, and your IDE probably doesn't either.
By having annotation, docs add it automatically when generating, and you don't need to read source.